src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,146 @@ void VulnerabilityScannerFacade::start(
         auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        // Queues to process skipped events
+        m_deltaQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(DELTA_EVENTS_QUEUE_PATH));
+        m_syncQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(SYNCHRONIZATION_EVENTS_QUEUE_PATH));
+
+        // Threads to retry skipped events
+        m_deltaThread = std::thread(
+            [scanOrchestrator, deltaQueue = m_deltaQueue, &stopRetryThread = m_stopRetryThread]()
+            {
+                while (!stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    while (!deltaQueue->empty() && !stopRetryThread.load())
+                    {
+                        std::shared_ptr<std::string> returnStr = deltaQueue->pop();
```

Comment:
The `pop()` method could return `nullptr` during the shutdown and this would cause a segmentation fault below at `returnStr->c_str()`.